api/src/main/java/org/apache/iceberg/io/ParallelIterable.java

```diff
@@ -28,25 +28,39 @@
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Future;
 import org.apache.iceberg.exceptions.RuntimeIOException;
-import org.apache.iceberg.io.CloseableGroup;
-import org.apache.iceberg.io.CloseableIterable;
-import org.apache.iceberg.io.CloseableIterator;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 import org.apache.iceberg.relocated.com.google.common.collect.Iterables;
 
+/**
+ * Run iterables in parallel.
+ * @deprecated please use {@link CloseableIterable#combine(Iterable, ExecutorService, int)} instead.
+ */
+@Deprecated
 public class ParallelIterable<T> extends CloseableGroup implements CloseableIterable<T> {
   private final Iterable<? extends Iterable<T>> iterables;
   private final ExecutorService workerPool;
+  private final int workerPoolSize;
 
+  /**
+   * @deprecated please use {@link CloseableIterable#combine(Iterable, ExecutorService, int)} instead.
```

Comment:
You can omit "please" from documentation so that docs are direct and as short as possible.

Also, can you add "will be removed in 0.14.0"? We like to keep track of when things can be removed.